libslide/src/math/gcd_poly_zz.rs

```diff
@@ -0,0 +1,430 @@
+use crate::math::gcd;
+
+use core::cmp::{max, min};
+
+#[allow(unused)]
+type Poly = Vec<isize>;
+
+#[allow(unused)]
+type PolyPtrArg = [isize];
+
+#[allow(unused)]
+fn gcd_poly_zz_heu(f: Poly, g: Poly) -> Option<(Poly, Poly, Poly)> {
```

Comment:
Please add documentation describing this function, the paper it comes from, and what it returns. Better yet the return type could be a struct.